common/src/main/scala/org/apache/celeborn/common/metrics/source/AbstractSource.scala

```diff
@@ -79,12 +79,27 @@ abstract class AbstractSource(conf: CelebornConf, role: String)
 
   val applicationLabel = "applicationId"
 
+  val timerMetricsMap: ConcurrentHashMap.KeySetView[String, lang.Boolean] =
```

Comment:
As you only need the string array without duplicating the values, I think you can use ConcurrentLinkedQueue to gain better performance. The timer values can not be duplicated because the string contains the current timestamp.

Reply:
Done～